### DISCUSSION TITLE
definePageMeta is not defined error

### DISCUSSION THREAD

**Discussion by yabuking84** [nuxt/nuxt#16236] [Questions]
An error will show (definePageMeta is not defined...) when using script setup on a page

`<script setup>
definePageMeta({
  layout: "custom",
});
</script>`

![image](https://user-images.githubusercontent.com/48405194/149914087-d0d42636-b8bf-48e4-845b-34add5275639.png)

**Comment by danielroe** [comment]
Would you raise an issue with a reproduction? 🙏

**Reply by yabuking84** [reply]
I fixed the issue by deleting package-lock.json, .nuxt, node_module then reinstalled node_module. Not sure what happened. Should I still raise this issue? or just post an answer here?

**Reply by danielroe** [reply]
I suspect you were using a version of Nuxt that did not have this feature.

**Reply by drm-addoptio** [reply]
Having the same problem. Easy to reproduce. Just create a new nuxt 3 project with `npx nuxi init nuxt3-app` and try to use it e.g. in the App.vue file within a setup script. This really should be fixed urgently. The solution provided by yabuking84 did not work for me.

**Reply by danielroe** [reply]
If you don't have a pages directory, then there is no `vue-router` integration and no support for `definePageMeta`.

**Reply by drm-addoptio** [reply]
OK, could reproduce the problem. After adding a pages folder and a layouts folder, the problem still arises. Then, deleting node_modules, installing it again (in this case with yarn install), using yarn dev, it works.

**Reply by danielroe** [reply]
Great. Sounds like you just needed to upgrade to the version that has this feature.

**Reply by drm-addoptio** [reply]
No, it is more like you have to reinstall, after you added the pages folder. I tested it immediately after setting up a new project. So there can't be a difference in the versions. When using definePageMeta, and trying to give layout different values depending on whether it is mobile device or not, seems not to work. Isn't it possible to assign a variable?

**Reply by selfagency** [reply]
I'm having this same problem with a brand new install as well and deleting my lockfile, node_modules, and .nuxt folders did not resolve it.

**Comment by yabuking84** [comment]
Fixed the issue by deleting package-lock.json, .nuxt, node_module then reinstalled node_module and re-run npm run dev to recreate .nuxt.

**Reply by Oldwo1f** [reply]
This solution didnt worked for me.

I still have the error

**Reply by Moskovskiy** [reply]
And so do I...

**Reply by danielroe** [reply]
I'd be very happy to look into it if you can raise an issue with a reproduction.

**Reply by Moskovskiy** [reply]
https://github.com/nuxt/framework/issues/2920

**Reply by tvbird** [reply]
> 

+1

**Reply by Ravshanovich99** [reply]
+1

**Reply by herpster** [reply]
This does not work.

**Reply by petter-tchirkov** [reply]
worked lika a charm
